src/components/FetchDataButton/FetchDataButton.tsx

```diff
@@ -0,0 +1,48 @@
+
+import React, { useState } from 'react';
+import Button from '@mui/material/Button';
+
+export type OnDataReceivedCallback<T = any> = (
+  data: T | null,
+  error: Error | null
+) => void;
+
+interface Parameters<T>{
+  text: string,
+  endpoint: string,
+  onDataReceived: OnDataReceivedCallback<T>,
+}
+
+/**
+ * A Button which when clicked, will make a request to the specified endpoint, and then call the passed in callback after this request completes.
+ */
+export const FetchDataButton = <T,>({ text = 'Click me!', endpoint, onDataReceived }: Parameters<T>) => {
```

Comment:
Mostly putting up this PR now so this can be used by anyone. It's helpful during testing hitting endpoints